client_http.hpp

```diff
@@ -16,7 +16,16 @@ namespace SimpleWeb {
   using errc = std::errc;
   using system_error = std::system_error;
   namespace make_error_code = std;
-  using string_view = const std::string &; // TODO c++17: use std::string_view
+#ifdef __has_include 
+  #if __has_include(<string_view>)
+    #include<string_view>
+    using string_view = std::string_view;
+    #define __has_string_view 1
+  #endif
+#endif
+#ifndef __has_string_view
+  using string_view = const std::string &;
+#endif
 } // namespace SimpleWeb
```

Comment:
I think these lines can be replaced with:
```c++
#if defined(__has_include) && __has_include(<string_view>)
  #include <string_view>
  using string_view = std::string_view;
#else
  using string_view = const std::string &;
#endif
```
Not tested though:) Ah, and the `#include <string_view>` has to be outside the namespace I think. The result could be:
```c++
#if defined(__has_include) && __has_include(<string_view>)
#include <string_view>
namespace SimpleWeb { using string_view = std::string_view; }
#elif defined(USE_STANDALONE_ASIO)
namespace SimpleWeb { using string_view = const std::string &; }
#else
namespace SimpleWeb { using string_view = boost::string_ref; }  
#endif
```
Can put this above line 11 for instance.

Reply:
@eidheim That should work. I had your first version originally and it compiled under MSVC, but I noted that IBMs example was fully expanded, and wondered if there was a reason for that. 

The version you have proposed makes sense, it would appear that I ignored the `USE_STANDALONE_ASIO` flag.